File: pyg-seckill-web/src/main/webapp/js/service/seckillGoodsService.js
app.service('seckillGoodsService',function($http){
	//读取列表数据绑定到表单中
	this.findList=function(){
		return $http.get('seckillGoods/findList.do');
	}
	
	//根据ID查询商品
	this.findOne=function(id){
		return $http.get('seckillGoods/findOneFromRedis.do?id='+id);
	}
	
	
	//提交订单
	this.submitOrder=function(seckillId){
		return $http.get('seckillOrder/submitOrder.do?seckillId='+seckillId);
	}

	//从缓存中移除
	this.removeGoodsFromRedis=function(goodsid){
		return $http.get('seckillGoods/removeGoodsFromRedis.do?goodsId='+goodsid);
	}
	//获取当前登录账号的收货地址
	this.findAddressList=function(){
		return $http.get('address/findListByLoginUser.do');
	}

});

File: pyg-seckill/pyg-seckill-interface/src/main/java/top/ingxx/seckill/service/SeckillGoodsService.java
package top.ingxx.seckill.service;

import top.ingxx.entity.PageResult;
import top.ingxx.pojo.TbSeckillGoods;
import top.ingxx.untils.entity.PygResult;

import java.util.List;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SeckillGoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbSeckillGoods> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbSeckillGoods seckillGoods);
	
	
	/**
	 * 修改
	 */
	public void update(TbSeckillGoods seckillGoods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbSeckillGoods findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbSeckillGoods seckillGoods, int pageNum, int pageSize);
	
	/**
	 * 返回正在参与秒杀的商品
	 * @return
	 */
	public List<TbSeckillGoods> findList();
	
	
	/**
	 * 根据ID获取实体(从缓存中读取)
	 * @param id
	 * @return
	 */
	public TbSeckillGoods findOneFromRedis(Long id);
	/**
	 * 当秒杀时间结束时从缓存中移除该商品
	 */
	public PygResult removeGoodsFromRedis(Long goodsid);
}
